lib/rubocop/ast/node.rb

```diff
@@ -528,6 +525,14 @@ def range_type?
         GROUP_FOR_TYPE[type] == :range
       end
 
+      def type_or_group_type?(type)
+        type == self.type || GROUP_FOR_TYPE[self.type] == type
+      end
```

Comment:
I'm uncertain whether the behavior indicated by the name `type_or_group_type?` is clear. Since it currently seems to be called only from `yield_for_types`, how about not publishing `type_or_group_type?` and making it a private method instead?

Reply:
I took another look and seems like I just reimplemented the existing `type?` method. So, I removed both of these new methods instead.